caddyhttp/httpserver/context_test.go

```diff
@@ -302,6 +303,41 @@ func TestIP(t *testing.T) {
 	}
 }
 
+type myIP string
+
+func (ip myIP) mockInterfaces() ([]net.Addr, error) {
+	a := net.ParseIP(string(ip))
+
+	return []net.Addr{
+		&net.IPNet{IP: a, Mask: nil},
+	}, nil
+}
+
+func TestServerIP(t *testing.T) {
+	context := getContextOrFail(t)
+
+	tests := []string{
+		// Test 0 - ipv4
+		"1.1.1.1",
+		// Test 1 - ipv6
+		"2001:db8:a0b:12f0::1",
+	}
+
+	for i, testIP := range tests {
+		testPrefix := getTestPrefix(i)
+
+		// Fake the network interfaces
+		ip := myIP(testIP)
+		context.NIFn = ip.mockInterfaces
```

Comment:
Once you make this a package variable, just set this at the top of the TestServerIP function and use `defer func() { ... }` to set it back to what it was (just as a nice-thing-to-do, keeping your tests clean and have fewer side-effects in case something else would be changed by it).